src/Workspaces/Core/Portable/Workspace/Workspace_Events.cs

```diff
@@ -59,22 +78,38 @@ protected Task RaiseWorkspaceChangedEventAsync(WorkspaceChangeKind kind, Solutio
             projectId = documentId.ProjectId;
         }
 
-        var ev = GetEventHandlers<WorkspaceChangeEventArgs>(WorkspaceChangeEventName);
+        var ev = GetEventHandlers<WorkspaceChangeEventArgs>(WorkspaceChangedImmediateEventName);
+        WorkspaceChangeEventArgs args = null;
+        if (ev.HasHandlers)
+        {
+            args = new WorkspaceChangeEventArgs(kind, oldSolution, newSolution, projectId, documentId);
```

Comment:
You could just pull this assignment out of the if() block here; practically any workspace where we would care about allocations will have at least one handler. :smile:

Reply:
Done!